pkg/controllers/status/cluster_status_controller.go

```diff
@@ -402,132 +413,104 @@ func getAPIEnablements(clusterClient *util.ClusterClient) ([]clusterv1alpha1.API
-
-	resourceSummary := &clusterv1alpha1.ResourceSummary{}
-	resourceSummary.Allocatable = allocatable
-	resourceSummary.Allocating = allocating
-	resourceSummary.Allocated = allocated
+func (c *ClusterStatusController) handleEvent(fedKey keys.FederatedKey) error {
+	runtimeObj, err := helper.GetObjectFromCache(c.RESTMapper, c.InformerManager, fedKey)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			c.ClusterSummaryCache.delete(fedKey)
+		}
+		return err
+	}
 
-	return resourceSummary
+	if err := c.ClusterSummaryCache.set(fedKey, runtimeObj); err != nil {
+		klog.Errorf("Failed to sync %s(%s) into cache, err is: %v", fedKey.Kind, fedKey.NamespaceKey(), err)
+		return err
+	}
+	return nil
 }
 
-func convertObjectsToNodes(nodeList []runtime.Object) ([]*corev1.Node, error) {
-	nodes := make([]*corev1.Node, 0, len(nodeList))
-	for _, obj := range nodeList {
-		node, err := helper.ConvertToNode(obj.(*unstructured.Unstructured))
-		if err != nil {
-			return nil, fmt.Errorf("failed to convert unstructured to typed object: %v", err)
-		}
-		nodes = append(nodes, node)
+// getEventHandler return callback function that knows how to handle events from the member cluster.
+func (c *ClusterStatusController) getEventHandler(clusterName string) cache.ResourceEventHandler {
+	if value, exists := c.eventHandlers.Load(clusterName); exists {
+		return value.(cache.ResourceEventHandler)
 	}
-	return nodes, nil
+
+	eventHandler := informermanager.NewHandlerOnEvents(c.genHandlerAddFunc(clusterName), c.genHandlerUpdateFunc(clusterName),
+		c.genHandlerDeleteFunc(clusterName))
+	c.eventHandlers.Store(clusterName, eventHandler)
+	return eventHandler
 }
 
-func convertObjectsToPods(podList []runtime.Object) ([]*corev1.Pod, error) {
-	pods := make([]*corev1.Pod, 0, len(podList))
-	for _, obj := range podList {
-		pod, err := helper.ConvertToPod(obj.(*unstructured.Unstructured))
+func (c *ClusterStatusController) genHandlerAddFunc(clusterName string) func(obj interface{}) {
+	return func(obj interface{}) {
+		curObj := obj.(runtime.Object)
+		key, err := keys.FederatedKeyFunc(clusterName, curObj)
 		if err != nil {
-			return nil, fmt.Errorf("failed to convert unstructured to typed object: %v", err)
+			klog.Warningf("Failed to generate key for obj: %s", curObj.GetObjectKind().GroupVersionKind())
+			return
 		}
-		pods = append(pods, pod)
+		c.worker.Add(key)
 	}
-	return pods, nil
 }
 
-func getClusterAllocatable(nodeList []*corev1.Node) (allocatable corev1.ResourceList) {
-	allocatable = make(corev1.ResourceList)
-	for _, node := range nodeList {
-		for key, val := range node.Status.Allocatable {
-			tmpCap, ok := allocatable[key]
-			if ok {
-				tmpCap.Add(val)
-			} else {
-				tmpCap = val
+func (c *ClusterStatusController) genHandlerUpdateFunc(clusterName string) func(oldObj, newObj interface{}) {
+	return func(oldObj, newObj interface{}) {
+		curObj := newObj.(runtime.Object)
+		if !reflect.DeepEqual(oldObj, newObj) {
```

Comment:
How do you think compare with resource version?

Reply:
I think it's ok to comparing with obj, because we don't know the type of obj. If comparing with resource version, may add the cost of converting type.

Reply:
I test it on my side with go 1.17.5
```go

var (
	obj1 interface{}
	obj2 interface{}
)

func init() {
        pod1Yaml := `<skipped>`
        pod2Yaml := `<skipped>`

	unstructuredObj1 := &unstructured.Unstructured{}
	unstructuredObj2 := &unstructured.Unstructured{}
	unstructured.UnstructuredJSONScheme.Decode([]byte(pod1Yaml), nil, unstructuredObj1)
	unstructured.UnstructuredJSONScheme.Decode([]byte(pod2Yaml), nil, unstructuredObj2)

        obj1, obj2 = unstructuredObj1, unstructuredObj2
}

func BenchmarkDeepEqual(b *testing.B) {
	for i := 0; i < b.N; i++ {
		_ = reflect.DeepEqual(obj1, obj2)
	}
}

func BenchmarkCast(b *testing.B) {
	for i := 0; i < b.N; i++ {
		o1, _ := meta.Accessor(obj1)
		o2, _ := meta.Accessor(obj2)
		_ = o1.GetResourceVersion() == o2.GetResourceVersion()
	}
}
```

```
goos: darwin
goarch: arm64
pkg: github.com/ikaven1024/gists/k8s/benchmark
BenchmarkDeepEqual
BenchmarkDeepEqual-8   	 7717909	       145.2 ns/op
BenchmarkCast
BenchmarkCast-8        	33085802	        36.39 ns/op
PASS
```

Type cast is more faster than `reflect.DeepEqual`